src/modules/flight_mode_manager/tasks/Transition/FlightTaskTransition.cpp

```diff
@@ -120,3 +108,123 @@ bool FlightTaskTransition::update()
+{
+	_sub_vehicle_status.update();
+	_sub_position_sp_triplet.update();
+	_sub_vehicle_local_position.update();
+}
+
+bool FlightTaskTransition::isVtolFrontTransition() const
+{
+	return _sub_vehicle_status.get().in_transition_mode
+	       && _sub_vehicle_status.get().in_transition_to_fw;
+
+}
+
+bool FlightTaskTransition::isVtolBackTransition() const
+{
+	return _sub_vehicle_status.get().in_transition_mode
+	       && !_sub_vehicle_status.get().in_transition_to_fw;
+}
+
+float FlightTaskTransition::computeBackTranstionPitchSetpoint()
+{
+	const vehicle_local_position_s &local_pos = _sub_vehicle_local_position.get();
+	const position_setpoint_s &current_pos_sp = _sub_position_sp_triplet.get().current;
+
+	// Retrieve default decelaration setpoint
+	const float default_deceleration_sp = _param_vt_b_dec_mss;
+	// Maximum allowed deceleration setpoint as a function of the nominal deceleration setpoint
+	const float max_deceleration_sp = 2.f * default_deceleration_sp;
+
+	float deceleration_sp = default_deceleration_sp;
+
+	const float track = atan2f(local_pos.vy, local_pos.vx);
+	const float accel_body_forward = cosf(track) * local_pos.ax + sinf(track) * local_pos.ay;
+	const float vel_body_forward = cosf(track) * local_pos.vx + sinf(track) * local_pos.vy;
+
+
+	if (current_pos_sp.valid && local_pos.xy_valid) {
+
+		float pos_sp_x, pos_sp_y = 0.f;
+		_geo_projection.project(current_pos_sp.lat, current_pos_sp.lon, pos_sp_x,
+					pos_sp_y);
+
+		// Compute backtransition end-point w.r.t. vehicle
+		const float pos_sp_dx = pos_sp_x - local_pos.x;
+		const float pos_sp_dy = pos_sp_y - local_pos.y;
+
+		// Compute the deceleration setpoint if the backtransition end-point is ahead of the vehicle
+		const float vel_proj = local_pos.vx * pos_sp_dx + local_pos.vy * pos_sp_dy;
+
+		if (vel_proj > 0.0f) {
+			const float dist_body_forward = cosf(track) * pos_sp_dx + sinf(track) * pos_sp_dy;
+
+			if (fabsf(dist_body_forward) > FLT_EPSILON) {
+				// Compute deceleration setpoint
+				// Note this is deceleration (i.e. negative acceleration), and therefore the minus sign is skipped
+				deceleration_sp = vel_body_forward * vel_body_forward / (2.f * dist_body_forward);
+
+				// Check if the deceleration setpoint is finite and within limits
+				if (!PX4_ISFINITE(deceleration_sp)) {
+					deceleration_sp = default_deceleration_sp;
+
+				} else {
+					// Limit the deceleration setpoint
+					deceleration_sp = math::constrain(deceleration_sp, 0.0f, max_deceleration_sp);
+				}
+			}
+		}
+	}
+
+	float pitch_sp_new = _decel_error_bt_int;
+
+	// Update deceleration error integrator
+	// get accel error, positive means decelerating too slow, need to pitch up (must reverse accel_body_forward, as it is a positive number)
+	float deceleration_error = deceleration_sp - (-accel_body_forward);
+
+	updateBackTransitioDecelerationErrorIntegrator(deceleration_error);
+
+	return math::constrain(pitch_sp_new, 0.f, _pitch_limit_bt);;
```

Comment:
_decel_error_bt_int is already constraint, this additional constraint is not necessary is it?

Reply:
totally right